crates/weaver_forge/src/v2/span.rs

```diff
@@ -0,0 +1,81 @@
+//! Span related definitions structs.
+
+use crate::v2::attribute::Attribute;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use weaver_semconv::{
+    attribute::RequirementLevel,
+    group::SpanKindSpec,
+    v2::{signal_id::SignalId, span::SpanName, CommonFields},
+};
+
+/// The definition of a span signal.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct Span {
+    /// The type of the Span. This denotes the identity
+    /// of the "shape" of this span, and must be unique.
+    pub r#type: SignalId,
+    /// Specifies the kind of the span.
+    pub kind: SpanKindSpec,
+    /// The name pattern for the span.
+    pub name: SpanName,
+    /// List of attributes that should be included on this span.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub attributes: Vec<SpanAttribute>,
+    /// Which resources this span should be associated with.
+    ///
+    /// This list is an "any of" list, where a span may be associated with one or more entities, but should
+    /// be associated with at least one in this list.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub entity_associations: Vec<String>,
+
+    /// Common fields (like brief, note, annotations).
+    #[serde(flatten)]
+    pub common: CommonFields,
+}
+
+/// A special type of reference to attributes that remembers span-specicific information.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct SpanAttribute {
+    /// Base attribute definitions.
+    #[serde(flatten)]
+    pub base: Attribute,
+    /// Specifies if the attribute is mandatory. Can be "required",
+    /// "conditionally_required", "recommended" or "opt_in". When omitted,
+    /// the attribute is "recommended". When set to
+    /// "conditionally_required", the string provided as <condition> MUST
+    /// specify the conditions under which the attribute is required.
+    ///
+    /// Note: For attributes that are "recommended" or "opt-in" - not all metric source will
+    /// create timeseries with these attributes, but for any given timeseries instance, the attributes that *were* present
+    /// should *remain* present. That is - a metric timeseries cannot drop attributes during its lifetime.
+    pub requirement_level: RequirementLevel,
+
+    /// Specifies if the attribute is (especially) relevant for sampling
+    /// and thus should be set at span start. It defaults to false.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub sampling_relevant: Option<bool>,
```

Comment:
there were some discussions on 'sampling-relevant' being non-intuitive name. Maybe we should rename it in v2 to something else? (We can also follow-up since this would go beyond the scope of this PR).

Copy-pasting from mast meeting notes:

> Rename to something like ‘head-sampling’  / ‘provided - at span start time’ 
"start span operation" <- specification term.

E.g. `record_on_start`, `provided_on_start`, `available_at_start_time`, etc

Reply:
Yeah - I think we need to make some decisions here.  I was looking in the past, and "sampling relevant" was still marked "experimental" on our old schema, but we somehow dropped that notion as we evolved into weaver.